.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,36 @@
+name: clang-tidy
+
+on:
+  push: {}
```

Comment:
on pull request would be good too to catch regressions

Reply:
could you add this to the change? then i can merge it.

Reply:
Done, and what do you think about enabling other clang-tidy checks? Because there is like 455 of them in total, and here I enabled only one class of them (clang static analyzer) that has 113 checks. Some other checks/groups might be nice to have

Reply:
starting small but passing is fine.

we can always enable more as we go :)